src/azul/indexer/document.py

```diff
@@ -396,8 +400,19 @@ def translate_fields(cls,
                 return [cls.translate_fields(val, field_types, forward=forward, path=path) for val in doc]
             else:
                 assert False, (path, type(doc))
-        elif isinstance(field_types, FieldType):
-            field_type = field_types
+        elif isinstance(field_types, (list, FieldType)):
+            if isinstance(field_types, list):
+                field_type = one(field_types)
+                assert isinstance(field_type, FieldType)
+
+                # In english: if forward and doc is not None, then if
+                # field_types is a list, doc should also be a list. We would
+                # also like to assert that a non-list field_type implies a
+                # non-list doc. Such an assertion could only be done for
+                # contributions, and only after https://github.com/DataBiosphere/azul/issues/2689
+                assert not forward or doc is None or isinstance(doc, list)
```

Comment:
Why do we assert this only in forward mode?

Reply:
The assert should apply in both directions. It required a small tweak to make this work.